### PR TITLE
Q&M: Fix broken links identified by the Broken links report - fix aspnetcore.docs repo

### PR DESCRIPTION
**Pull request by Daidihuang**
Hi @cmastr, @Rick-Anderson, @wadepickett, @tdykstra I have fixed the broken links, please verify, thanks。
https://dev.azure.com/msft-skilling/Content/_workitems/edit/335393/?view=edit

---

#### Internal previews

| 📄 File | 🔗 Preview link |
|:--|:--|
| [aspnetcore/blazor/components/dynamiccomponent.md](https://github.com/dotnet/AspNetCore.Docs/blob/6573c89094f0aa5f733c2be7591952b22dd20c35/aspnetcore/blazor/components/dynamiccomponent.md) | [aspnetcore/blazor/components/dynamiccomponent](https://review.learn.microsoft.com/en-us/aspnet/core/blazor/components/dynamiccomponent?branch=pr-en-us-34001) |
| [aspnetcore/blazor/components/layouts.md](https://github.com/dotnet/AspNetCore.Docs/blob/6573c89094f0aa5f733c2be7591952b22dd20c35/aspnetcore/blazor/components/layouts.md) | [aspnetcore/blazor/components/layouts](https://review.learn.microsoft.com/en-us/aspnet/core/blazor/components/layouts?branch=pr-en-us-34001) |
| [aspnetcore/blazor/host-and-deploy/configure-linker.md](https://github.com/dotnet/AspNetCore.Docs/blob/6573c89094f0aa5f733c2be7591952b22dd20c35/aspnetcore/blazor/host-and-deploy/configure-linker.md) | [aspnetcore/blazor/host-and-deploy/configure-linker](https://review.learn.microsoft.com/en-us/aspnet/core/blazor/host-and-deploy/configure-linker?branch=pr-en-us-34001) |
| [aspnetcore/blazor/progressive-web-app.md](https://github.com/dotnet/AspNetCore.Docs/blob/6573c89094f0aa5f733c2be7591952b22dd20c35/aspnetcore/blazor/progressive-web-app.md) | [aspnetcore/blazor/progressive-web-app](https://review.learn.microsoft.com/en-us/aspnet/core/blazor/progressive-web-app?branch=pr-en-us-34001) |
| [aspnetcore/blazor/security/webassembly/graph-api.md](https://github.com/dotnet/AspNetCore.Docs/blob/6573c89094f0aa5f733c2be7591952b22dd20c35/aspnetcore/blazor/security/webassembly/graph-api.md) | [aspnetcore/blazor/security/webassembly/graph-api](https://review.learn.microsoft.com/en-us/aspnet/core/blazor/security/webassembly/graph-api?branch=pr-en-us-34001) |
| [aspnetcore/blazor/security/webassembly/standalone-with-identity.md](https://github.com/dotnet/AspNetCore.Docs/blob/6573c89094f0aa5f733c2be7591952b22dd20c35/aspnetcore/blazor/security/webassembly/standalone-with-identity.md) | [aspnetcore/blazor/security/webassembly/standalone-with-identity](https://review.learn.microsoft.com/en-us/aspnet/core/blazor/security/webassembly/standalone-with-identity?branch=pr-en-us-34001) |
| [aspnetcore/grpc/supported-platforms.md](https://github.com/dotnet/AspNetCore.Docs/blob/6573c89094f0aa5f733c2be7591952b22dd20c35/aspnetcore/grpc/supported-platforms.md) | [aspnetcore/grpc/supported-platforms](https://review.learn.microsoft.com/en-us/aspnet/core/grpc/supported-platforms?branch=pr-en-us-34001) |
| [aspnetcore/release-notes/aspnetcore-8.0.md](https://github.com/dotnet/AspNetCore.Docs/blob/6573c89094f0aa5f733c2be7591952b22dd20c35/aspnetcore/release-notes/aspnetcore-8.0.md) | [aspnetcore/release-notes/aspnetcore-8.0](https://review.learn.microsoft.com/en-us/aspnet/core/release-notes/aspnetcore-8.0?branch=pr-en-us-34001) |
| [aspnetcore/test/dev-tunnels.md](https://github.com/dotnet/AspNetCore.Docs/blob/6573c89094f0aa5f733c2be7591952b22dd20c35/aspnetcore/test/dev-tunnels.md) | [aspnetcore/test/dev-tunnels](https://review.learn.microsoft.com/en-us/aspnet/core/test/dev-tunnels?branch=pr-en-us-34001) |

### CODE REVIEW

**Comment by guardrex** [comment]
The Blazor link changes require a few updates. I'll get to it *ASAP* 🏃‍♂️.

File: aspnetcore/blazor/components/dynamiccomponent.md
Comment (guardrex) [inline aspnetcore/blazor/components/dynamiccomponent.md:552]:
```suggestion
For a working demonstration of the preceding example, see the [`DynamicComponentExample4` component in the Blazor sample app](https://github.com/dotnet/blazor-samples/blob/main/7.0/BlazorSample_Server/Pages/dynamiccomponent/DynamicComponentExample4.razor).
```

File: aspnetcore/blazor/components/layouts.md
Comment (guardrex) [inline aspnetcore/blazor/components/layouts.md:84]:
```suggestion
* [Blazor Web App `MainLayout.razor.css`](https://github.com/dotnet/aspnetcore/blob/main/src/ProjectTemplates/Web.ProjectTemplates/content/BlazorWeb-CSharp/BlazorWeb-CSharp/Components/Layout/MainLayout.razor.css)
```

File: aspnetcore/release-notes/aspnetcore-8.0.md
Comment (guardrex) [inline aspnetcore/release-notes/aspnetcore-8.0.md:266]: No loc segments in links as long as the loc segment is redirected (added) at the destination ...

```suggestion
Blazor now supports the [`cancel`](https://developer.mozilla.org/docs/Web/API/HTMLDialogElement/cancel_event) and [`close`](https://developer.mozilla.org/docs/Web/API/HTMLDialogElement/close_event) events on the `dialog` HTML element.
```

**Review by guardrex — CHANGES_REQUESTED** [review]
Thanks @Daidihuang ... I've updated the Blazor links. I'm going to go ahead and commit them now. I'll take care of that one in the progressive Blazor app article (Line 125) on that other issue that I opened.

**Comment by guardrex** [comment]
BTW @Daidihuang ... Are you a vendor or a community dev? Either way, I suppose that you'll need to accept the Contributor License Agreement. Otherwise, we won't be able to merge this PR.

**Comment by Daidihuang** [comment]
*   I am making Submissions in the course of work for my employer (or my employer has intellectual property rights in my Submissions by contract or applicable law). I have permission from my employer to make Submissions and enter into this Agreement on behalf of my employer. By signing below, the defined term “You” includes me and my employer.

     @dotnet-policy-service agree company="Microsoft"

________________________________

**Comment by Daidihuang** [comment]
I am making Submissions in the course of work for my employer (or my employer has intellectual property rights in my Submissions by contract or applicable law). I have permission from my employer to make Submissions and enter into this Agreement on behalf of my employer. By signing below, the defined term “You” includes me and my employer.

@dotnet-policy-service agree company="Microsoft"




Thanks

Daidi Huang

C+E Skilling - Pubdesk - Beyondsoft Team

Team Email: ***@***.******@***.***> | Service Hour: 8:00 – 17:00 PST

C+E Skilling Publishing Desk Service<https://review.docs.microsoft.com/en-us/content-production-service/topics/pubdesk-documentation/pubdesk-internal/?branch=main>

________________________________

**Review by guardrex — APPROVED** [review]
@Rick-Anderson ... I think this is fine now. I'll work on the one more-complicated Blazor link as soon as my Internet is back up. I'll leave this to u for a final check and merge.

**Comment by guardrex** [comment]
@Rick-Anderson ... I'll go ahead with this to unblock the other issue that I opened.

File: aspnetcore/test/dev-tunnels.md
Comment (guardrex) [inline aspnetcore/test/dev-tunnels.md:169]:
```suggestion
* [ACS Call Automation and Azure OpenAI Service](https://github.com/Azure-Samples/communication-services-dotnet-quickstarts/tree/main/callautomation-openai-sample-csharp#acs-call-automation-and-azure-openai-service)

**Pull request merged**
